assets/src/setup/components/unsaved-changes-warning/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Options } from '../options-context-provider';
+
+/**
+ * If there are unsaved changes in the wizard, warns the user before exiting the page.
+ *
+ * @return {null} Renders nothing.
+ */
+export function WizardUnsavedChangesWarning() {
+	const { hasChanges, hasSaved } = useContext( Options );
+
+	useEffect( () => {
+		if ( ! hasChanges || hasSaved ) {
```

Comment:
Is this correct? Shouldn't it be the other way around?

```suggestion
		if ( hasChanges || ! hasSaved ) {
```

Reply:
Yes, had to push an additional commit to get it right, but good catch. Thanks!